clang/lib/AST/ExprConstant.cpp

```diff
@@ -7806,7 +7806,8 @@ class ExprEvaluatorBase
       // Overloaded operator calls to member functions are represented as normal
       // calls with '*this' as the first argument.
       const CXXMethodDecl *MD = dyn_cast<CXXMethodDecl>(FD);
-      if (MD && MD->isImplicitObjectMemberFunction()) {
+      if (MD &&
+          (MD->isImplicitObjectMemberFunction() || (OCE && MD->isStatic()))) {
```

Comment:
So in the static case, what value do we expect `ThisVal` to have and does that have consequences that the code after this is prepared for?

Reply:
It doesn't look like anything has gone wrong so far, but just to be on the safe side, maybe we could change the code below to this?

```diff
- This = &ThisVal;
+ if (MD->isInstance())
+   This = &ThisVal;
```